src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,273 @@
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 2.00f;
+            case KC_METRO:
+            case SOUND_TRANSIT:
+            case EVERETT_TRANSIT:
+            case SEATTLE_STREET_CAR:
+                return 1.50f;
+            default: return defaultFare;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        List<Ride> ridesSingleton = new ArrayList<>();
+        ridesSingleton.add(ride);
+        return calculateCost(fareType, ridesSingleton, fareRules);
+    }
+
+    /**
+     * Calculate the cost of a journey. Where free transfers are not permitted the cash price is use. If free transfers
+     * are applicable, the most expensive discount fare across all legs is added to the final cumulative cash price.
+     */
+    @Override
+    public boolean populateFare(Fare fare,
+                                   Currency currency,
+                                   Fare.FareType fareType,
+                                   List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules
+    ) {
+        float cost = 0;
+        float orcaFareDiscount = 0;
+        for (Ride ride : rides) {
+            RideType rideType = classify(ride.routeData);
+            float singleLegPrice = getRidePrice(ride, fareType, fareRules);
+            float discountedFare = getDiscountedFare(fareType, rideType, singleLegPrice);
+            if (hasFreeTransfers(fareType, rideType)) {
+                // If using Orca (free transfers), the total fare should be equivalent to the
+                // most expensive leg of the journey.
+                orcaFareDiscount = Float.max(orcaFareDiscount, discountedFare);
+            } else {
+                // If free transfers not permitted (i.e., paying with cash), accumulate each
+                // leg as we go.
+                cost += singleLegPrice;
+            }
+        }
+        cost += orcaFareDiscount;
+        if (cost < Float.POSITIVE_INFINITY) {
+            fare.addFare(fareType, getMoney(currency, cost));
+        }
+        return cost > 0 && cost < Float.POSITIVE_INFINITY;
+    }
+
+    /**
+     * If using Orca and the ride type is not Washington state ferries (they do no allow free transfers) a free transfer
+     * can be applied.
+     */
+    private boolean hasFreeTransfers(Fare.FareType fareType, RideType rideType) {
+        return rideType != RideType.WASHINGTON_STATE_FERRIES && usesOrca(fareType);
```

Comment:
Skagit Transit also does not allow free transfers, but check with @jon-campbell-ibigroup about whether we should include logic for that agency in this PR or at a later time.

Reply:
@evansiroky I've asked the question in the comm trans channel.

Reply:
I've not heard back, but have added it.